bcda/models/postgres/alr.go

```diff
@@ -0,0 +1,252 @@
+package postgres
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"encoding/gob"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/CMSgov/bcda-app/bcda/models"
+	"github.com/jackc/pgx"
+	"github.com/jackc/pgx/stdlib"
+	"github.com/sirupsen/logrus"
+)
+
+/*******************************************************************************
+	DATA STRUCTURES
+		1. AlrRepository
+		2. alrCopyFromSource
+			- Used to implement the CopyFrom from the pgx package CopyFrom is
+			Go's connection to PostgreSQL's COPY functionality.
+*******************************************************************************/
+
+type AlrRepository struct {
+	*sql.DB
+	*pgx.Conn
+}
+
+type alrCopyFromSource struct {
+	ctx       context.Context
+	metaKey   int
+	timestamp time.Time
+	alrCount  int
+	alrRowLen int
+	alrs      []models.Alr
+}
+
+/*******************************************************************************
+	HELPER FUNCTIONS
+		1. NewAlrRepo - Used to instantiate AlrRepository struct
+		2. gogEncoder
+			- Used for turning map[string]string to []bytes
+		3. gobDecoder
+			- Used for turning []bytes back to map
+*******************************************************************************/
+
+func NewAlrRepo(db *sql.DB) *AlrRepository {
+	conn, err := stdlib.AcquireConn(db)
+	if err != nil {
+		logrus.Warn(err, "failed to acquire connection")
+	}
+	return &AlrRepository{db, conn}
+}
+
+func GobEncoder(mp map[string]string) ([]byte, error) {
```

Comment:
Do the encoder and decoder need to be visible outside of this package?

And would `func encode` / `func decode` better describe the function